certora/specs/ERC721.spec

```diff
@@ -185,29 +176,73 @@ invariant ownedTotalIsSumOfBalances()
 
 /*
 ┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
-│ Invariant: balanceOf is the number of tokens owned                                                                  │
+│ Invariant: balanceOf matches the ghost _balances mapping                                                            │
 └─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
 */
 invariant balanceOfConsistency(address user)
-    to_mathint(balanceOf(user)) == _ownedByUser[user] &&
     to_mathint(balanceOf(user)) == _balances[user]
     {
         preserved {
             require balanceLimited(user);
+            
+            // ERC721 specific constraints to prevent underflow
+            require to_mathint(balanceOf(user)) >= 0;
```

Comment:
`balanceOf(user)` is a uint256, so by definition it should always be `>=0`. This looks like a totology